src/api/store/supported_language.py

```diff
@@ -0,0 +1,77 @@
+from database.models import SupportedLanguage
+from _main_.utils.massenergize_errors import CustomMassenergizeError
+from _main_.utils.activity_logger import log
+from typing import Tuple
+from django.db import IntegrityError
+
+log = log.exception
+
+class SupportedLanguageStore:
+    def __init__ (self):
+        self.name = "Supported Language Store/DB"
+
+    def get_supported_language_info (self, args) -> Tuple[ dict or None, any ]:
+        try:
+            language_code = args.get('language_code', None)
+
+            if not language_code:
+                return None, CustomMassenergizeError("Please provide a valid id or code")
```

Comment:
same feedback here.  centralize your error messages somewhere and reference them here.
Also its also good for this particular case that we log what that invalid code is.
```suggestion
                return None, CustomMassenergizeError("Please provide a valid id or code: {0}".format(language_code)
```